packages/salesforcedx-vscode-core/src/commands/forceRenameLightningComponent.ts

```diff
@@ -0,0 +1,140 @@
+    placeHolder: nls.localize(RENAME_INPUT_PLACEHOLDER),
+    promopt: nls.localize(RENAME_INPUT_PROMPT)
+  } as vscode.InputBoxOptions;
+
+  const responseText = await vscode.window.showInputBox(inputOptions);
+  if (sourceFsPath) {
+    const commandlet = new SfdxCommandlet(
+      new SfdxWorkspaceChecker(),
+      new FilePathGatherer(sourceUri),
+      new RenameLwcComponentExecutor(sourceFsPath, responseText)
+    );
+    await commandlet.run();
+  }
+}
+
+function renameComponent(sourceFsPath: string, newName: string) {
+  const componentPath = getComponentPath(sourceFsPath);
+  const componentName = path.basename(componentPath);
+  checkForDuplicateName(componentPath, newName);
+  const items = fs.readdirSync(componentPath);
+  for (const item of items) {
+    // only rename the file that has same name with component
+    if (isNameMatch(item, componentName, componentPath)) {
+      const newItem = item.replace(componentName, newName);
+      fs.renameSync(
+        path.join(componentPath, item),
+        path.join(componentPath, newItem)
+      );
+    }
+  }
+  const newComponentPath = path.join(path.dirname(componentPath), newName);
+  fs.renameSync(
+    componentPath,
+    newComponentPath
+  );
+  notificationService.showWarningMessage(nls.localize(RENAME_WARNING));
+}
+
+function getComponentPath(sourceFsPath: string): string {
+  const stats = fs.statSync(sourceFsPath);
+  return stats.isFile() ? path.dirname(sourceFsPath) : sourceFsPath;
+}
+
+function checkForDuplicateName(componentPath: string, newName: string) {
+  if (isDuplicate(componentPath, newName)) {
+    const errorMessage = nls.localize(RENAME_INPUT_DUP_ERROR);
+    notificationService.showErrorMessage(errorMessage);
+    throw new Error(format(errorMessage));
+  }
+}
+
+function isDuplicate(componentPath: string, newName: string): boolean {
+  // A LWC component can't share the same name as a Aura component
+  const componentPathDirName = path.dirname(componentPath);
+  let lwcPath: string;
+  let auraPath: string;
+  if (isLwcComponent(componentPath)) {
+    lwcPath = componentPathDirName;
+    auraPath = path.join(path.dirname(componentPathDirName), 'aura');
+  } else {
+    lwcPath = path.join(path.dirname(componentPathDirName), 'lwc');
+    auraPath = componentPathDirName;
+  }
+  const allLwcComponents = fs.readdirSync(lwcPath);
+  const allAuraComponents = fs.readdirSync(auraPath);
+  if (allLwcComponents.includes(newName) || allAuraComponents.includes(newName)) {
+    return true;
+  }
+  return false;
+}
+
+function isNameMatch(item: string, componentName: string, componentPath: string) {
+  const isLwc = isLwcComponent(componentPath);
+  let regularExp: RegExp;
+  if (isLwc) {
+    regularExp = new RegExp(`${componentName}\.(html|js|js-meta.xml|css|svg)`);
+  } else {
+    regularExp = new RegExp(`${componentName}(((Controller|Renderer|Helper)?\.js)|(\.(cmp|app|css|design|auradoc|svg)))`);
+  }
+  return item.match(regularExp) ? true : false;
```

Comment:
does item.match return true/false?  If so just return it.

Reply:
@gbockus-sf item.match does not return true/false but returns an array that includes matched string,

Reply:
gotcha...returns null if no match.  I still think `return Boolean(item.match(regularExp));` is more readable.